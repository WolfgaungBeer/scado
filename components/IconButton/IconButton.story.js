import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs';
import IconButton from './';

storiesOf('IconButton', module)
    .add('default', () => (
        <IconButton icon="check_circle" onClick={action('clicked')} />
    ))
    .add('with props', () => (
        <IconButton
            icon={text('icon', 'check_circle')}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            loading={boolean('loading', false)}
            onClick={action('clicked')}
        />
    ));
