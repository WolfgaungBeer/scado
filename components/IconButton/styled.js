import styled from 'styled-components';

export const StyledButton = styled.button`
    cursor: pointer;
    background-color: transparent;
    border-style: none;
    &:focus {
        outline: 0;
    }
`;
