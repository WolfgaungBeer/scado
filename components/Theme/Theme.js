import React from 'react';
import { shape, object, node } from 'prop-types';
import { ThemeProvider } from 'styled-components';

const propTypes = {
    theme: shape({
        screen: object,
        color: object,
        scale: object,
    }).isRequired,
    children: node,
};

const defaultProps = {
    children: undefined,
};

const Theme = ({ theme, children }) => (<ThemeProvider theme={theme}>{children}</ThemeProvider>);

Theme.propTypes = propTypes;
Theme.defaultProps = defaultProps;

export default Theme;
