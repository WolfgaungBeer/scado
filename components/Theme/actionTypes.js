export const SET_THEME = '@SCADO/THEME/SET_THEME';
