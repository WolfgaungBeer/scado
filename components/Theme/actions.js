import types from './actionTypes';

const setTheme = payload => ({
    type: types.SET_THEME,
    payload,
});

export default {
    setTheme,
};
