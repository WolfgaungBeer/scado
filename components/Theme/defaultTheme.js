export default {

    color: {
        black: '#202021',
        white: '#F7F7F7',
        info: '#48C5FC',
        success: '#37B661',
        warning: '#DD9913',
        error: '#E25650'
    },

    scale: {
        xs: '0.25rem',
        s: '0.5rem',
        m: '1rem',
        l: '1.5rem',
        xl: '2rem',
        xxl: '2.5rem',
        xxxl: '3rem'
    }

}
