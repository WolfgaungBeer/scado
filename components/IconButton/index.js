import React from 'react';
import { string, bool, func } from 'prop-types';
import { StyledButton } from './styled';
import Icon from '../Icon';

const propTypes = {
    icon: string.isRequired,
    color: string,
    scale: string,
    loading: bool,
    onClick: func.isRequired,
};

const defaultProps = {
    color: undefined,
    scale: undefined,
    loading: undefined,
};

const IconButton = ({ icon, color, scale, loading, onClick }) => (
    <StyledButton onClick={onClick}>
        <Icon icon={icon} color={color} scale={scale} loading={loading} />
    </StyledButton>
);

IconButton.propTypes = propTypes;
IconButton.defaultProps = defaultProps;

export default IconButton;
