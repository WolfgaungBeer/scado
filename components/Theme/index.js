import { connect } from 'react-redux';
import { theme } from './selectors';
import Theme from './Theme';

const mapStateToProps = (state) => {
    return {
        theme: theme(state)
    };
};

export default connect(mapStateToProps)(Theme);
