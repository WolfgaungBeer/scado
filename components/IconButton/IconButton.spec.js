import React from 'react';
import renderer from 'react-test-renderer';
import Theme from '../Theme/Theme';
import defaultTheme from '../Theme/defaultTheme';
import IconButton from './';

const click = () => {};

it('renders the Icon Component correctly', () => {
    const tree1 = renderer.create(
        <Theme theme={defaultTheme}>
            <IconButton icon="check_circle" onClick={click} />
        </Theme>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Theme theme={defaultTheme}>
            <IconButton icon="check_circle" color="success" scale="l" onClick={click} />
        </Theme>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Theme theme={defaultTheme}>
            <IconButton icon="check_circle" color="success" scale="l" loading onClick={click} />
        </Theme>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();
});
