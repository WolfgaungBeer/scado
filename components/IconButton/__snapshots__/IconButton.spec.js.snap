// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Icon Component correctly 1`] = `
<button
  className="sc-bdVaJa hOjKLo"
  onClick={[Function]}
>
  <i
    className="material-icons  sc-bwzfXH kYzDSj"
    color={undefined}
    scale={undefined}
  >
    check_circle
  </i>
</button>
`;

exports[`renders the Icon Component correctly 2`] = `
<button
  className="sc-bdVaJa hOjKLo"
  onClick={[Function]}
>
  <i
    className="material-icons  sc-bwzfXH jhuqqZ"
    color="success"
    scale="l"
  >
    check_circle
  </i>
</button>
`;

exports[`renders the Icon Component correctly 3`] = `
<button
  className="sc-bdVaJa hOjKLo"
  onClick={[Function]}
>
  <i
    className="material-icons  sc-bwzfXH dqDpaz"
    color="success"
    scale="l"
  >
    autorenew
  </i>
</button>
`;
